### PR TITLE
[serve] Make monitoring section top-level in the docs

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

This is currently nested under "deployment," which can be hard to find (a few users weren't able to find it).

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
Basically purely a refactor, no need to review content (only changed the top-level description of each page)

File: doc/source/serve/deployment.rst [hunk 1]
```diff
@@ -309,266 +309,3 @@ Feel free to open new github issues and contribute more storage backends!
 
 .. _serve-monitoring:
```
Comment (simon-mo) [inline doc/source/serve/deployment.rst:310]: need to move this reference as well. 

and also can we add a "Moved! notes here to redirect user to new section"

**Review by simon-mo — APPROVED** [review]

File: doc/source/serve/deployment.rst (hunk 1 above)
Comment (edoakes) [reply doc/source/serve/deployment.rst:310]: ah yeah will move the reference

i don't think we need that, they can just search for it

Comment (simon-mo) [reply doc/source/serve/deployment.rst]: ah right, forgot that our search is good now

**Pull request merged**